src/MLJDecisionTreeInterface.jl

```diff
@@ -392,52 +357,325 @@ function MMI.predict(::RandomForestRegressor, forest, Xnew)
+
+- `min_samples_leaf=1`:    max number of samples each leaf needs to have
+
+- `min_samples_split=2`:   min number of samples needed for a split
+
+- `min_purity_increase=0`: min purity needed for a split
+
+- `n_subfeatures=0`: number of features to select at random (0 for all,
+  -1 for square root of number of features)
+
+- `post_prune=false`:      set to `true` for post-fit pruning
+
+- `merge_purity_threshold=1.0`: (post-pruning) merge leaves having
+                           combined purity `>= merge_purity_threshold`
+
+- `rng=Random.GLOBAL_RNG`: random number generator or seed
+
+
+### Operations
+
+- `predict(mach, Xnew)`: return predictions of the target given new
+  features `Xnew` having the same scitype as `X` above.
+
+
+### Fitted parameters
+
+The fields of `fitted_params(mach)` are:
+
+- `tree`: the tree or stump object returned by the core
+  DecisionTree.jl algorithm
+
+See also
+[DecisionTree.jl](https://github.com/bensadeghi/DecisionTree.jl) and
+the unwrapped model type
+[`MLJDecisionTreeInterface.DecisionTree.DecisionTreeRegressor`](@ref).
+
+"""
+
+const DOC_RANDOM_FOREST_REGRESSOR = """
+$(MMI.doc_header(RandomForestRegressor))
+
+### Training data
+
+In MLJ or MLJBase, bind an instance `model` to data with
+
+    mach = machine(model, X, y)
+
+where
+
+- `X`: any table of input features (eg, a `DataFrame`) whose columns
+  each have one of the following element scitypes: `Continuous`,
+  `Count`, or `<:OrderedFactor`.
+
+- `y`: the target, which can be any `AbstractVector` whose element
+  scitype is `Continuous`.
+
+Train the machine with `fit!(mach, rows=...)`.
+
+
+### Hyper-parameters
+
+- `max_depth=-1`:          max depth of the decision tree (-1=any)
+
+- `min_samples_leaf=1`:    min number of samples each leaf needs to have
+
+- `min_samples_split=2`:   min number of samples needed for a split
+
+- `min_purity_increase=0`: min purity needed for a split
+
+- `n_subfeatures=-1`: number of features to select at random (0 for all,
+  -1 for square root of number of features)
+
+- `n_trees=10`:            number of trees to train
+
+- `sampling_fraction=0.7`  fraction of samples to train each tree on
+
+- `rng=Random.GLOBAL_RNG`: random number generator or seed
+
+
+### Operations
+
+- `predict(mach, Xnew)`: return predictions of the target given new
+  features `Xnew` having the same scitype as `X` above.
+
+
+### Fitted parameters
+
+The fields of `fitted_params(mach)` are:
+
+- `forest`: the `Ensemble` object returned by the core DecisionTree.jl algorithm
+
+See also
+[DecisionTree.jl](https://github.com/bensadeghi/DecisionTree.jl) and
+the unwrapped model type
+[`MLJDecisionTreeInterface.DecisionTree.RandomForestRegressor`](@ref).
+
+"""
+
+@doc DOC_RANDOM_FOREST_CLASSIFIER RandomForestClassifier
+@doc DOC_ADA_BOOST_STUMP_CLASSIFIER AdaBoostStumpClassifier
+@doc DOC_DECISION_TREE_REGRESSOR DecisionTreeRegressor
+@doc DOC_RANDOM_FOREST_REGRESSOR RandomForestRegressor
```

Comment:
Why is chosen for using `@doc` instead of putting the string above the struct definition?

Reply:
But have cleaned this up a bit.